pulp_container/app/serializers.py

```diff
@@ -265,25 +265,30 @@ def validate(self, data):
                         "repository version."
                     )
                 )
-        base_path = validated_data["base_path"]
-        namespace_name = base_path.split("/")[0]
 
-        if "namespace" in validated_data:
-            if validated_data["namespace"].name != namespace_name:
-                raise serializers.ValidationError(
-                    _("Selected namespace does not match first component of base_path.")
+        if "base_path" in validated_data:
+            base_path = validated_data["base_path"]
+            namespace_name = base_path.split("/")[0]
+
+            if "namespace" in validated_data:
+                if validated_data["namespace"].name != namespace_name:
+                    raise serializers.ValidationError(
+                        _("Selected namespace does not match first component of base_path.")
+                    )
+            else:
+                validated_data["namespace"] = ContainerNamespaceSerializer.get_or_create(
+                    {"name": namespace_name}
                 )
-        else:
-            validated_data["namespace"] = ContainerNamespaceSerializer.get_or_create(
-                {"name": namespace_name}
-            )
+        elif self.partial is False:
```

Comment:
what corner case are we covering here? base_path is a required field to my knowledge

Reply:
I tried to do a partial update to change "private" and it blew up, because "base_path" wasn't part of the `validated_data`.
This elif branch is maybe a safetynet. It should never be reached. I agree. Maybe i should raise an Implementation Error of some kind instead of a ValidationError to get a proper 500 server response.
WDYT?

Reply:
if it is a patch request, why do we enforce users to provide base_path if i want to only update the 'name', for example.
can we just leave the if branch on line 269 without the elif branch on line 282?

Reply:
That's what i wanted to say if drf is implemented correctly, 282 should never be reached. I may have been to cautious. Removing it now...